### PR TITLE
mom hooks getting "global name 'pbs_connect' is not defined" error

### PR DESCRIPTION
**Pull request by agrawalravi90**
#### Bug/feature Description
In particular, if the hook calls "pbs.server()", it fails with the message:

03/04/2019 01:27:58;0001;pbs_python;Svr;pbs_python;PBS server internal error (15011) in Error evaluating Python script, <type 'exceptions.NameError'>
03/04/2019 01:27:58;0001;pbs_python;Svr;pbs_python;PBS server internal error (15011) in Error evaluating Python script, **global name 'pbs_connect' is not defined**

#### Affected Platform(s)
* All

#### Cause / Analysis / Design
* This happened because of #988 where the IFL functions were changed to macros
* The hook infrastructure uses SWIG converted PBS C functions. It needs to call the PBS C IFL functions like pbs_connect(), but since they were converted to macros, the IFL function definitions in SWIG generated modules disappeared.

#### Solution Description
* Convert the PBS IFL macros back to regular functions which call the function pointers which were added in #988 .

#### Testing logs/output
* Running any test which created a hook that called pbs.server() would be good enough to test this change. I ran TestAdminSuspend's test_hook() test:
[test_hook_after.log](https://github.com/PBSPro/pbspro/files/3019036/test_hook_after.log)
[test_hook_before.log](https://github.com/PBSPro/pbspro/files/3019037/test_hook_before.log)
* Will generate a build on Windows soon to make sure that it still works there.

#### Checklist:

- [X] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [X] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [X] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [X] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [X] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/include/pbs_ifl.h [hunk 1]
```diff
@@ -581,51 +581,182 @@ DECLDIR char *__pbs_server_location(void);
...
+
+extern char *pbs_default(void);
+
+extern int pbs_deljob(int, char *, char *);
+
+extern char *pbs_geterrmsg(int);
+
+extern int pbs_holdjob(int, char *, char *, char *);
+
+extern int pbs_loadconf(int);
```
Comment (bayucan) [inline src/include/pbs_ifl.h]: Check but I think on the OSS include/ifl_internal.h, there's also a similar extern declaration to pbs_loadconf()...

File: src/lib/Libifl/ifl_impl.c [hunk 2]
```diff
@@ -0,0 +1,788 @@
...
+ *
+ * @param[in] c - connection handler
+ * @param[in] preempt_jobs_list - list of jobs to be preempted
+ *
+ * @return      int
+ * @retval      0       success
+ * @retval      !0      error
+ *
+ */
+preempt_job_info*
```
Comment (bhroam) [inline src/lib/Libifl/ifl_impl.c]: You should probably be consistent with other pointer return values and put a space between the type and pointer.

**Comment by agrawalravi90** [comment]
@bhroam and @bayucan Thanks for the review! I've addressed your comments, please let me know if things look okay now.

**Comment by agrawalravi90** [comment]
Marked this WIP because I am going to do a full regression and also test this on Windows, but please do continue the review, thanks!

**Comment by agrawalravi90** [comment]
I'm reasonably confident in these changes, so request you to merge this @bhroam , thanks in advance!

**Comment by bhroam** [comment]
@agrawalravi90 please do one rebase/squash and retest.  If another one is required, I'll take it from there.

**Comment by agrawalravi90** [comment]
@bhroam I rebased and tested this again, test logs: 
[test_hook.log](https://github.com/PBSPro/pbspro/files/3040567/test_hook.log)

**Comment by bhroam** [comment]
Thanks @agrawalravi90.  I merged in the PR.

**Pull request merged**